src/hooks/useSarcophagusNegotiation.ts

```diff
@@ -77,36 +99,45 @@ export function useSarcophagusNegotiation() {
                 // TODO: remove these logs after we gain some confidence in this exchange
                 console.log('got', dataStr);
 
-                const { signature }: { signature: string } = JSON.parse(dataStr);
-                console.log('setting arch signature');
-                console.log(signature);
-                archaeologistSignatures.set(arch.profile.archAddress, signature);
+                const response = JSON.parse(dataStr);
+                if (response.error) {
+                  dispatch(setArchaeologistException(arch.profile.peerId, {
+                    code: ArchaeologistExceptionCode.DECLINED_SIGNATURE,
+                    message: processDeclinedSignatureCode(response.error.code as SarcophagusValidationError, arch.profile.archAddress),
+                  }));
+                } else {
+                  archaeologistSignatures.set(arch.profile.archAddress, response.signature);
+
+                  dispatch(setArchaeologistSignature(
+                    arch.profile.peerId,
+                    response.signature
+                  ));
+                }
               }
-            }).finally(() => {
-              stream.close();
-            });
-          })
-        );
-
-        setArchaeologistSignatures(archaeologistSignatures);
-      } catch (err) {
-        //TODO figure out what to do at this point
-        console.error(`Error in initiateSarcophagusNegotiation: ${err}`);
-      }
+            }).catch(e => {
+              // TODO: `message` will (likely) be user-facing. Need friendlier verbiage.
+              const message = `Exception occurred in negotiaton stream for: ${arch.profile.archAddress}`;
+              console.error(`Stream exception on ${arch.profile.peerId}`, e);
+              dispatch(setArchaeologistException(arch.profile.peerId, {
+                code: ArchaeologistExceptionCode.STREAM_EXCEPTION,
+                message
+              }));
+            }).finally(() => stream.close());
+          } catch (e) {
+            // TODO: `message` will (likely) be user-facing. Need friendlier verbiage.
+            const message = `Exception occurred attempting to open stream to: ${arch.profile.archAddress}`;
+            console.error(`Stream exception on ${arch.profile.peerId}`, e);
+            dispatch(setArchaeologistException(arch.profile.peerId, {
+              code: ArchaeologistExceptionCode.CONNECTION_EXCEPTION,
```

Comment:
This catch should only ever be thrown from `newStream` above (94). It's reasonably a connection type exception if that fails.